File: 046_OCaml/fizzbuzz.ml
(* This is a comment. *)

let fizzbuzz i =
    match (i mod 3), (i mod 5) with
        0, 0 -> "FizzBuzz" |
        0, _ -> "Fizz" |
        _, 0 -> "Buzz" |
        _    -> string_of_int i

let () =
    for i = 1 to 100 do
        print_endline @@ fizzbuzz i (* `A @@ B` is equivalent to `A (B)`. *)
    done

File: 033_GAS/fizzbuzz.js
// This is a comment.

function fizzbuzz() {
    for (let i = 1; i <= 100; ++i) {
        if (i % 3 == 0 && i % 5 == 0) {
            Logger.log('FizzBuzz');
        } else if (i % 3 == 0) {
            Logger.log('Fizz');
        } else if (i % 5 == 0) {
            Logger.log('Buzz');
        } else {
            Logger.log(i.toFixed(0));
        }
    }
}

